### PR TITLE
refactor: simplify native token address / zero address

### PR DESCRIPTION
**Pull request by septs**
closes #

### CODE REVIEW

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx [hunk 1]
```diff
@@ -73,7 +71,7 @@ export const TransferERC20 = memo<TransferERC20Props>(({ token }) => {
     }, [token])
 
     // workaround: transferERC20 should support non-evm network
-    const isNativeToken = isSameAddress(selectedToken?.address, NATIVE_TOKEN_ADDRESS)
+    const isNativeToken = isNative(selectedToken?.address)
     const tokenType = isNativeToken ? EthereumTokenType.Native : EthereumTokenType.ERC20
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:75]:
```suggestion
    const tokenType = isNative(selectedToken?.address) ? EthereumTokenType.Native : EthereumTokenType.ERC20
```

File: packages/mask/src/plugins/Trader/trader/0x/useTrade.ts [hunk 2]
```diff
@@ -18,18 +17,20 @@ import { currentNetworkSettings } from '../../../Wallet/settings'
 import { TargetChainIdContext } from '../useTargetChainIdContext'
 import { TradeProvider } from '@masknet/public-api'
 
+const NATIVE_TOKEN_LABEL = `0x${'e'.repeat(40)}`
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trader/0x/useTrade.ts:20]:
```suggestion
const ZRX_NATIVE_TOKEN_ADDRESS = `0x${'e'.repeat(40)}`
```

File: packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx (hunk 1 above)
Comment (septs) [reply packages/dashboard/src/pages/Wallets/components/Transfer/TransferERC20.tsx:75]: CleanShot 2021-12-22 at 17 42 37@2x

File: packages/mask/src/extension/background-script/EthereumServices/tokenList.ts [hunk 3]
```diff
@@ -56,7 +57,7 @@ async function fetch1inchERC20TokensFromTokenList(
     const tokens = ((await fetchTokenList(url)) as TokenObject).tokens
     const _tokens = Object.values(tokens)
     return _tokens
-        .filter((x) => x.address.toLowerCase() !== NATIVE_TOKEN_ADDRESS_IN_1INCH)
+        .filter((x) => isSameAddress(x.address, NATIVE_TOKEN_ADDRESS_IN_1INCH))
```
Comment (guanbinrui) [inline packages/mask/src/extension/background-script/EthereumServices/tokenList.ts:60]:
```suggestion
        .filter((x) => !isSameAddress(x.address, NATIVE_TOKEN_ADDRESS_IN_1INCH))
```

File: packages/mask/src/plugins/ITO/SNSAdaptor/ConfirmDialog.tsx [hunk 4]
```diff
@@ -152,7 +152,7 @@ export function ConfirmDialog(props: ConfirmDialogProps) {
                         <Typography variant="body1" component="span">
                             {poolSettings?.token?.symbol}
                         </Typography>
-                        {isNative(poolSettings?.token?.address!) ? null : (
+                        {isZeroAddress(poolSettings?.token?.address!) ? null : (
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/ConfirmDialog.tsx:155]:
```suggestion
                        {isZeroAddress(poolSettings?.token?.address) ? null : (
```

File: packages/mask/src/plugins/ITO/SNSAdaptor/SwapGuide.tsx [hunk 5]
```diff
@@ -101,11 +100,7 @@ export function SwapGuide(props: SwapGuideProps) {
                         case SwapStatus.Unlock:
                             return (
                                 <UnlockDialog
-                                    tokens={
-                                        payload.exchange_tokens.filter(
-                                            (x) => !isSameAddress(NATIVE_TOKEN_ADDRESS, x.address),
-                                        ) as ERC20TokenDetailed[]
-                                    }
+                                    tokens={payload.exchange_tokens.filter(isZeroAddress) as ERC20TokenDetailed[]}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/SwapGuide.tsx:103]: Break change.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketConfirmDialog.tsx [hunk 6]
```diff
@@ -93,7 +92,7 @@ export function RedPacketConfirmDialog(props: ConfirmRedPacketFormProps) {
             <Grid item xs={6}>
                 <Typography variant="body1" color="textPrimary" align="right" className={classes.token}>
                     <span>{settings?.token?.symbol}</span>
-                    {isNative(settings?.token?.address!) ? null : (
+                    {isZeroAddress(settings?.token?.address!) ? null : (
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketConfirmDialog.tsx:95]:
```suggestion
                    {isZeroAddress(settings?.token?.address) ? null : (
```

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedpacketNftConfirmDialog.tsx [hunk 7]
```diff
@@ -232,7 +232,7 @@ export function RedpacketNftConfirmDialog(props: RedpacketNftConfirmDialogProps)
                             align="right"
                             className={classNames(classes.account, classes.bold, classes.text)}>
                             ({wallet?.name}) {formatEthereumAddress(account, 4)}
-                            {isNative(wallet?.address!) ? null : (
+                            {isZeroAddress(wallet?.address!) ? null : (
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedpacketNftConfirmDialog.tsx:235]:
```suggestion
                            {isZeroAddress(wallet?.address) ? null : (
```

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketConfirmDialog.tsx [hunk 8]
```diff
@@ -1,12 +1,11 @@
 import BigNumber from 'bignumber.js'
 import classNames from 'classnames'
 import { useEffect } from 'react'
-import { formatBalance, resolveTokenLinkOnExplorer, useChainId } from '@masknet/web3-shared-evm'
+import { formatBalance, isZeroAddress, resolveTokenLinkOnExplorer, useChainId } from '@masknet/web3-shared-evm'
 import { Grid, Link, Paper, Typography } from '@mui/material'
 import { makeStyles } from '@masknet/theme'
 import LaunchIcon from '@mui/icons-material/Launch'
 import { FormattedBalance } from '@masknet/shared'
-import { isNative } from 'lodash-unified'
```
Comment (guanbinrui) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketConfirmDialog.tsx:9]: What!

**Review by guanbinrui — DISMISSED** [review]

**Comment by nuanyang233** [comment]
Maybe we should move `isSameAddress` and `isZeroAddress` to `@masknet/shared/utils`

**Pull request closed without merging**